lib/shared/puppet_self_service.rb

```diff
@@ -121,4 +121,33 @@ def self.filesystem_free(path)
     stat = Sys::Filesystem.stat(path)
     (stat.blocks_available.to_f / stat.blocks.to_f * 100).to_i
   end
+
+  def self.NETHTTPpuppetAPI(puppetendpoint)
+
+    uri = URI.parse("#{puppetendpoint}")
```

Comment:
This should be using `Puppet::HTTP::Client` https://www.rubydoc.info/gems/puppet/Puppet/HTTP/Client and needs to rescue on exceptions.

Reply:
Made changes in #50 to utilize `Puppet::HTTP::Client` which uses Puppet certs by default so you may be able to use the "new new" method haha.